### PR TITLE
enable linting in project

### PR DESCRIPTION
**Pull request by marcdumais-work**
Signed-off-by: Marc Dumais <marc.dumais@ericsson.com>

#### What it does
This PR adds required configs so that linting (tslint) will work, as part of the TypeScript language server and report problems as-one-develops.

#### How to test
Open `.ts` files and intentionally break configured linting rules (e.g. 2 consecutive blank lines, remote semicolon at end of line). Verify that the linter report errors and that visual feedback is provided in the Theia editor.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,9 @@
     "lerna": "^2.2.0",
     "mocha": "^3.4.2",
     "nyc": "^11.0.3",
-    "tslint": "^5.12.0"
+    "tslint": "^5.12.0",
```
Comment (marcdumais-work) [inline package.json]: note: I follow the pattern we have in the main repo: an extension or example application should not dictate what linter / linter version or compiler/compiler version is used to develop. Rather these should be dictated by a given project's configuration. 

This is why `tslint` and `typescript` are defined in this project's root `package.json` and not in the extensions or example applications. This way if we use a the example app from this repo or one based on it, or a Theia application that contains e.g. @theia/cpp-debug, to develop in JS./TS project, we will not override their choices of compiler/linter.

**Comment by marcdumais-work** [comment]
@thegecko I know we'll likely soon remove `@cortex/debug` but in the meantime do you know if the `tsconfig` setting `"skipLibCheck": true` was useful/needed? We have this to `false` for the extensions in the Theia main repo and for `@theia/cpp-debug`.

I think that's the only setting that ends-up being changed with this PR, unless I introduced bugs.

**Comment by vince-fugnitto** [comment]
> @thegecko I know we'll likely soon remove `@cortex/debug` but in the meantime do you know if the `tsconfig` setting `"skipLibCheck": true` was useful/needed? We have this to `false` for the extensions in the Theia main repo and for `@theia/cpp-debug`.
> 
> I think that's the only setting that ends-up being changed with this PR, unless I introduced bugs.

I believe it is fine to have the `skipLibCheck: true`. All it says is not to worry about declaration problems from upstream. We advised some extension developers to use it when their build was failing due to declaration problems in the main repository.

**Comment by marcdumais-work** [comment]
> I believe it is fine to have the `skipLibCheck: true`.

ok, let's do this change separately if/when needed.

**Comment by marcdumais-work** [comment]
thanks @vince-fugnitto

**Comment by vince-fugnitto** [comment]
> > I believe it is fine to have the `skipLibCheck: true`.
> 
> ok, let's do this change separately if/when needed.

Sure, for more information we set it to `true` from the generator, so extenders don't need to worry too much about our typings. 

https://github.com/theia-ide/generator-theia-extension/blob/3a8afcc9277abddd597e4c2f98ccfa4ed1a60caa/templates/tsconfig.json#L3

**Pull request merged**